cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go

```diff
@@ -156,6 +161,53 @@ type gceMig struct {
 	maxSize    int
 }
 
+// InstanceStatus represents status of instance. It collapses more verbose list of statuses from
+// GCE to just three values (running, being created, being deleted)
+type InstanceStatus uint64
```

Comment:
Is there any reason not to use cloudprovider.InstanceState? They seem identical.

Reply:
Honestly all the data structures here look like a duplicate of stuff you already added at cloudprovider level. The only difference I see is distinguishing between quota and stockout and I'm not sure how valuable that is.